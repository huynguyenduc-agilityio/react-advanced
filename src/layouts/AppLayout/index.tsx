import { useLocation } from 'react-router-dom';
import { useEffect } from 'react';

import MainLayout from '@/layouts/MainLayout';

const AppLayout = () => {
  const location = useLocation();

  useEffect(() => {
    document.documentElement.scrollTo(0, 0);
  }, [location.pathname]);

  return <MainLayout />;
};

export default AppLayout;
