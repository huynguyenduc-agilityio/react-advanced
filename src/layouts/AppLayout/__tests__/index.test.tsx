import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

// Layouts
import AppLayout from '@/layouts/AppLayout';

jest.mock('@/layouts/MainLayout', () => () => (
  <div data-testid="main-layout" />
));

describe('AppLayout', () => {
  beforeAll(() => {
    Object.defineProperty(document.documentElement, 'scrollTo', {
      value: jest.fn(),
      writable: true,
    });
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AppLayout />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
