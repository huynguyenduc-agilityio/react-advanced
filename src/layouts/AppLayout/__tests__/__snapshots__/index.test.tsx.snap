// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="main-layout"
  />
</DocumentFragment>
`;
